aiida/orm/querybuilder.py

```diff
@@ -53,117 +53,211 @@
+    classifiers = {}
+
+    classifiers['process_type_string'] = None
+    classifiers['ormclass_type_string'] = ormclass_type_string.lower()
+
+    if classifiers['ormclass_type_string'] == 'group':
+        ormclass = qb.Group
+    elif classifiers['ormclass_type_string'] == 'computer':
+        ormclass = qb.Computer
+    elif classifiers['ormclass_type_string'] == 'user':
+        ormclass = qb.User
     else:
-        # At this point, it has to be a node. The only valid string at this point is a string
+        # At this point, we assume it is a node. The only valid type string then is a string
         # that matches exactly the _plugin_type_string of a node class
-        ormclass = obj.Node
-        plugin_type_string = ormclasstype
-        query_type_string = get_query_type_from_type_string(plugin_type_string)
+        classifiers['ormclass_type_string'] = ormclass_type_string  # no lowercase
+        ormclass = qb.Node
+
+    if ormclass == qb.Node:
+        is_valid_node_type_string(classifiers['ormclass_type_string'], raise_on_false=True)
 
-    return plugin_type_string, query_type_string, ormclass
+
+    return ormclass, classifiers
+
+
+def get_type_filter(classifiers, subclassing):
+    """
+    Return filter dictionaries given a set of classifiers.
+
+    :param classifiers: a dictionary with classifiers (note: does *not* support lists)
+    :param subclassing: if True, allow for subclasses of the ormclass
+
+    :returns: dictionary in QueryBuilder filter language to pass into {"type": ... }
+    :rtype: dict
+
+    """
+    from aiida.orm.utils.node import get_query_type_from_type_string
+    from aiida.common.escaping import escape_for_sql_like
+    value = classifiers['ormclass_type_string']
+
+    if not subclassing:
+        filter = {'==': value}
+    else:
+        filter = {'like': '{}%'.format(escape_for_sql_like(get_query_type_from_type_string(value)))}
+
+    return filter
+
+def get_process_type_filter(classifiers, subclassing):
+    """
+    Return filter dictionaries given a set of classifiers.
+
+    :param classifiers: a dictionary with classifiers (note: does *not* support lists)
+    :param subclassing: if True, allow for subclasses of the process type
+            This is activated only, if an entry point can be found for the process type
+            (as well as for a selection of built-in process types)
+
+
+    :returns: dictionary in QueryBuilder filter language to pass into {"process_type": ... }
+    :rtype: dict
+
+    """
+    from aiida.common.escaping import escape_for_sql_like
+    from aiida.common.warnings import AiidaEntryPointWarning
+    from aiida.engine.processes.process import get_query_string_from_process_type_string
+    import warnings
+
+    value = classifiers['process_type_string']
+
+    if not subclassing:
+        filter = {'==': value}
+    else:
+        if ":" in value:
+            # if value is an entry point, do usual subclassing
+            filter = {'or': [
```

Comment:
Why do we need an `or` statement again in the case of sub classing on the `process_type` but not in the case of the `type`? I forgot

Reply:
The `process_type` is stored in the database without a `.` at the end.

Reply:
Ah right, if you push the small changes for the other comments I will approve and merge

Reply:
I've addressed your comments now.
Your comment made me realize an issue - it seems to me one should use `get_query_type_from_type_string` also [here](https://github.com/aiidateam/aiida_core/pull/2421/files#diff-ee49942ec9ece0e5ff43f3572bb42843R205) but when adding it, some tests fail.
I currently don't understand why (perhaps I'm also mistaken). If it is an issue, I'll prefer to fix this in a separate PR.

Reply:
Yes, let's get this merged and address any issues that may arise later